modules/ROOT/pages/comments-callback-mode.adoc

```diff
@@ -6,23 +6,36 @@
 :plugincode: comments
 :commentsMode: callback
 
-*Callback mode* is the default mode for the {pluginname} plugin. In the callback mode, callback functions are required to save user comments on a server. The {pluginname} functions (create, reply, edit, delete comment, delete all conversations, resolve, lookup, and fetch conversations) are configured differently depending upon the server-side storage configuration.
+*Callback mode* is the default mode for the {pluginname} plugin. In callback mode, callback functions are required to save user comments on a server. The {pluginname} functions (create, reply, edit, delete comment, delete all conversations, resolve, lookup, and fetch conversations) are configured differently depending upon the server-side storage configuration.
+
+[NOTE]
+====
+Callback mode provides the most flexibility for integrating with custom backend systems. All comment operations are handled through custom callback functions that communicate with your server.
+====
 
 == How the {pluginname} plugin works in callback mode
 
-All options accept functions incorporating `+done+` and `+fail+` callbacks as parameters. The function return type is not important, but all functions must call exactly one of these two callbacks: `+fail+` or `+done+`.
+All callback functions accept three parameters: a request object (or array for `+tinycomments_fetch+`), a `+done+` callback, and a `+fail+` callback. The function return type is not important, but all functions must call exactly one of these two callbacks: `+fail+` or `+done+`.
 
-* The `+fail+` callback takes either a string or a JavaScript Error type.
-* The `+done+` callback takes an argument specific to each function.
+* The `+fail+` callback takes either a string or a JavaScript Error type to indicate an unrecoverable error.
```

Comment:
As a dev, I would like to know what the plugin does with the string or error.

What does an unrecoverable error mean?

Reply:
```suggestion
* The `+fail+` callback takes an argument specific to each function.
```